client/components/social-buttons/google-one-tap.js

```diff
@@ -0,0 +1,228 @@
+				if ( width > 0 && height > 0 ) {
+					setTimeout( () => {
+						const buttonSize = this.buttonRef.current.getBoundingClientRect();
+						const ratio = buttonSize.width / ( width - 20 );
+						iframe.style.transform = `scale(${ ratio })`;
+						iframe.style.marginLeft = `${ Math.floor( -10 * ratio ) }px`;
+						iframe.style.marginBottom = '20px';
+						setTimeout( () => {
+							this.setState( { googleOneTapLoaded: true } );
+						}, 100 );
+					}, 20 );
+				}
+			}
+		} );
+
+		resizeObserver.observe( iframe );
+	}
+
+	async loadGoogleIdentityServicesAPI() {
+		if ( ! window?.google?.accounts?.id ) {
+			try {
+				await loadScript( 'https://accounts.google.com/gsi/client' );
+			} catch {
+				// It's safe to ignore loading errors because if Google is blocked in some way the the button will be disabled.
+				return null;
+			}
+		}
+
+		return window?.google?.accounts?.id ?? null;
+	}
+
+	async handleAuthorizationCode( { auth_code, redirect_uri, state } ) {
+		let response;
+		try {
+			response = await postLoginRequest( 'exchange-social-auth-code', {
+				service: 'google',
+				auth_code,
+				redirect_uri,
+				client_id: config( 'wpcom_signup_id' ),
+				client_secret: config( 'wpcom_signup_key' ),
+				state,
+			} );
+		} catch ( httpError ) {
+			const { code: error_code } = getErrorFromHTTPError( httpError );
+
+			if ( error_code ) {
+				this.props.recordTracksEvent( 'calypso_social_button_auth_code_exchange_failure', {
+					social_account_type: 'google',
+					starting_point: this.props.startingPoint,
+					error_code,
+				} );
+			}
+
+			this.props.showErrorNotice(
+				this.props.translate(
+					'Something went wrong when trying to connect with Google. Please try again.'
+				)
+			);
+
+			return;
+		}
+
+		this.props.recordTracksEvent( 'calypso_social_button_auth_code_exchange_success', {
+			social_account_type: 'google',
+			starting_point: this.props.startingPoint,
+		} );
+
+		const { access_token, id_token } = response.body.data;
+
+		this.props.responseHandler( { access_token, id_token } );
+	}
+
+	async fetchNonceAndInitializeGoogleSignIn() {
+		try {
+			const response = await wpcomRequest( {
+				path: '/generate-authorization-nonce',
+				apiNamespace: 'wpcom/v2',
+				apiVersion: '2',
+				method: 'GET',
+			} );
+			const state = response.nonce;
+
+			await this.initializeGoogleSignIn( state );
+		} catch ( error ) {
+			this.props.showErrorNotice(
+				this.props.translate(
+					'Error fetching nonce or initializing Google sign-in. Please try again.'
+				)
+			);
+		}
+	}
+
+	render() {
+		const isDisabled = Boolean( this.props.isFormDisabled );
+		const { googleOneTapLoaded } = this.state;
+
+		return (
+			<>
+				<div
+					ref={ this.buttonRef }
+					className="google__one-tap-sign-in-container"
+					data-type="standard"
+					data-disabled={ isDisabled }
+					data-theme="outline"
+					data-text="sign_up_with"
+					data-shape="rectangular"
+					data-locale={ getLocaleSlug() }
```

Comment:
It seems like the locale is not properly applied.
https://developers.google.com/identity/gsi/web/reference/html-reference#data-locale
Looking over the docs it says that `iso-639-code` needs to be passed. 
I tried hardcoding and passing different locales but that didn't affect the logged-out state of the button. I get the same text there.
<img width="522" alt="Screenshot 2024-07-18 at 12 44 58" src="https://github.com/user-attachments/assets/172de45e-d09d-4f9d-8bfa-f9bfa4555a95">

Reply:
Yes, I covered that in `Locale issues` in the PR description. Not sure how to fix this one tbh.